### ISSUE TITLE
trial documentation says --coverage requires python 2.3.3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4616]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#4616|
|Type|defect|
|Created|2010-07-27 09:20:49Z|
|Branch|https://github.com/twisted/twisted/tree/remove-lineno-patch-4616|

I suppose it should be python 2.3.3 or higher or something like that.

Attachments:

* [trial.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/db5/db55c7b937ca2fe42eb5da291ef30f1e490a72c6/62acd7f03265eca5ce1dc22dc4463368ea42efb0.diff) (2430 bytes) - added by fijal on 2010-07-27 09:28:04Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4616 4616
type__defect defect
reporter__fijal fijal
priority__low low
milestone__ 
branch__branches_remove_lineno_patch_4616 branches/remove-lineno-patch-4616
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1280222449000000 1280222449000000
changetime__1284124658000000 1284124658000000
version__None None
owner__ 
cc__fijal
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)<a name="note_1"></a>|@fijal commented|
|-|-|

Well, twisted does not support python 2.3 any more.

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj set owner to @fijal|
|-|-|

The code in question doesn't seem to have tests (`twisted/scripts/test`) and the patch doesn't come with any tests. I can see at least one NameError (the `trace` import was removed but code still uses the module.) Think you could add some?

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_5"></a>|@jonathanj commented|
|-|-|

The docstring seems to be missing a noun or something. Screwtape also points out that the docstring refers to `_trial_temp` explicitly, but the root path is obviously determined by `--temp-directory`. Perhaps something like: Write coverage information to a directory named "coverage", beneath the trial temporary directory.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

This news file should probably just be a `.misc`.  End users don't care about the code being removed because no _functionality_ is being removed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [29928]) Branching to 'remove-lineno-patch-4616'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner|
|-|-|

I added a test for `opt_coverage`.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-lineno-patch-4616).  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_9"></a>|@jml set owner to @exarkun|
|-|-|

Looks great! Thanks. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [29991]) Merge remove-lineno-patch-4616

Author: fijal, exarkun
Reviewer: jonathanj, jml
Fixes: [#4616](https://github.com/twisted/twisted/issues/4616)

Remove the trace module monkey patching from the implementation of
trial's `--coverage` option.  It was only necessary on certain
point releases of Python 2.3, no longer supported.  Also add a test
verifying that a trace hook is actually installed by `opt_coverage`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `reopened`|
|-|-|

(In [29992]) Revert r29991 - Python 2.4 test failure

```
[ERROR]
Traceback (most recent call last):
  File "/home/buildslave/slaves/twisted/bot-idnar-debian64/full2.4-debian64/Twisted/twisted/trial/test/test_script.py", line 402, in setUp
    self.addCleanup(sys.settrace, sys.gettrace())
exceptions.AttributeError: 'module' object has no attribute 'gettrace'

twisted.trial.test.test_script.CoverageTests.test_tracerInstalled
```

Reopens [#4616](https://github.com/twisted/twisted/issues/4616)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [29995]) Skip this test if there is no sys.gettrace

refs [#4616](https://github.com/twisted/twisted/issues/4616)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [29996]) Merge remove-lineno-patch-4616

Author: fijal, exarkun
Reviewer: jonathanj, jml
Fixes: [#4616](https://github.com/twisted/twisted/issues/4616)

Remove the trace module monkey patching from the implementation of
trial's `--coverage` option.  It was only necessary on certain
point releases of Python 2.3, no longer supported.  Also add a test
verifying that a trace hook is actually installed by `opt_coverage`.


Re-merge fixes test failure on Python 2.4, which does not have
`sys.gettrace` (the test is simply skipped in that case, as there
is no way to inspect the installed trace hook without `gettrace`).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
